test/suite/test_rollback_to_stable15.py

```diff
@@ -0,0 +1,133 @@
+        # Variable length
+        ('variable', dict(value_format='i')),
+    ]
+    scenarios = make_scenarios(key_format_values, value_format_values)
+
+    def check(self, check_value, uri, nrows, read_ts):
+        session = self.session
+        if read_ts == 0:
+            session.begin_transaction()
+        else:
+            session.begin_transaction('read_timestamp=' + timestamp_str(read_ts))
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, check_value)
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def test_rollback_to_stable(self):
+        # Create a table.
+        uri = "table:tmp"
+        nrows = 2000
+        create_params = 'key_format={},value_format={}'.format(self.key_format, self.value_format)
+        self.session.create(uri, create_params)
+        cursor =  self.session.open_cursor(uri)
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        value20 = 0x20
+        value30 = 0x30
+        value30 = 0x40
+        value40 = 0x50
+
+        #Insert value20 at timestamp 2
+        for i in range(1, nrows): # make a variable.
+            self.session.begin_transaction()
+            cursor[i] = value20
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(2))
+
+        #First Update to value 30 at timestamp 5
+        for i in range(1, nrows):
+            self.session.begin_transaction()
+            cursor[i] = value30
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(5))
+
+        #Set stable timestamp to 2
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(2))
+        self.conn.rollback_to_stable()
+        # Check that only value20 is available
+        self.check(value20, uri, nrows - 1, 2)
+
+        #Second Update to value30 at timestamp 7
+        for i in range(1, nrows):
+            self.session.begin_transaction()
+            cursor[i] = value30
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(7))
+
+        #Third Update to value40 at timestamp 9
+        for i in range(1, nrows):
+            self.session.begin_transaction()
+            cursor[i] = value40
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(9))
+
+        #Set stable timestamp to 7
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(7))
+        self.conn.rollback_to_stable()
+        #Check that only value30 is available
+        self.check(value30, uri, nrows - 1, 7)
+
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        upd_aborted = stat_cursor[stat.conn.txn_rts_upd_aborted][2]
+        keys_restored = stat_cursor[stat.conn.txn_rts_keys_restored][2]
```

Comment:
As per the above test, there shouldn't be any `keys_restored`. You can confirm it by checking that value as `0`.

Reply:
So would it be best to remove that line of code?

Reply:
Yes, removing is also fine.